sockshop/catalogue/cmd/cataloguesvc/main.go

```diff
@@ -52,16 +47,23 @@ func main() {
 	}
 
 	// Data domain.
-	socks, tags, err := readFile(*file)
+	db, err := sqlx.Open("mysql", *dsn)
 	if err != nil {
 		logger.Log("err", err)
 		os.Exit(1)
 	}
+	defer db.Close()
+
+	// Check if DB connection can be made, only for logging purposes, should not fail/exit
+	err = db.Ping()
+	if err != nil {
+		logger.Log("Error", "Unable to connect to Database", "DSN", dsn)
+	}
 
 	// Service domain.
 	var service catalogue.Service
 	{
-		service = catalogue.NewFixedService(socks, tags)
+		service = catalogue.NewCatalogueService(db, logger)
```

Comment:
Useful to know the pattern and how to use the library. Agree that in this case we don't need it (yet).